### PR TITLE
Add a "Vendor Supported" quality metric

### PR DESCRIPTION
**Pull request by mattray**
We need a means of highlighting cookbooks provided and supported by other companies for their products.

### CODE REVIEW

File: quality-metrics/new/qm-000-vendor_supported.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+---
+
+# The cookbook is supported by the vendor
+
+The vendor of the product or project managed by the cookbook acknowledges and
+accepts support responsibilities for the cookbook.
+
+* The vendor is listed as a contact email address for the cookbook in the README
+* Chef's Partner Engineering is listed as a contact email address for the cookbook in the README
+* Supermarket represents the cookbook as "Vendor Supported" in some form (badge?)
```
Comment (nathenharvey) [inline quality-metrics/new/qm-000-vendor_supported.md:15]: This is likely separate from the quality metric.  It's probably best as an issue / feature request for chef/supermarket.  I could see a new "partner manager" role on the Supermarket that gives the users in the role the ability to mark a cookbook as 'vendor supported' or something along those lines.  That mark could be used for searching and for display.

File: quality-metrics/new/qm-000-vendor_supported.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    end
+
+### Points
+
+* Positive Points: 1
+* Negative Points: 0
+
+There is no penalty for not being a partner-supported cookbook and this may be
+most appropriately presented as metadata for the cookbook to be used for sorting
+instead of scoring.
```
Comment (nathenharvey) [inline quality-metrics/new/qm-000-vendor_supported.md:40]: We've decided against implementing any sort of scoring for the first round so you can remove the points section of this PR.

**Comment by nathenharvey** [comment]
Love the idea but since there won't be any way to automatically detect this I think it's best implemented as a new feature for the Supermarket separate from the quality metrics.

**Comment by iennae** [comment]
Agreed with Nathen. As a reflection of the quality metrics, is it something that we would want people to aspire to have vendor support? I think it's separate information that we want to share and visualize on the  Supermarket, but not something that gives a cookbook additional quality. Do we have a specific set of practices we expect to happen based on vendor supported cookbooks?

**Comment by martinb3** [comment]
Just another thought -- we should clarify that vendor support may not be a binary yes/no. It may be that vendors support cookbooks that are unrelated to their product (e.g. if you pay us, we'll help you with any cookbook), vendors providing support for a cookbook that directly deals with their product (e.g. an oracle_java cookbook that installs java), as well as a vendor that only supports a cookbook for paid/licensed users (e.g. oracle_rdbms cookbook that installs the Oracle enterprise edition database).

I suggest we limit the scope of 'vendor supported' to be:
- the cookbook installs software supported by a vendor
- the vendor also supports the cookbook itself

Should we also make any kind of statement about vendor support that's been orphaned (no responses in X time period)? E.g. a cookbook the vendor _used_ to support, and the docs say the vendor still does, but they clearly don't?

**Comment by nathenharvey** [comment]
This is being implemented in the Supermarket so I'm going to close this issue.

Reference:
- https://github.com/chef/supermarket/issues/1175
- https://github.com/chef/supermarket/issues/1176
- https://github.com/chef/supermarket/issues/1177
- https://github.com/chef/supermarket/issues/1178

**Comment by martinb3** [comment]
I thought "vendor supported" was more general than Chef partner supported. Or was "chef partner support" the original intent of this issue? :8ball:

**Comment by mattray** [comment]
@martinb3 "partner supported" is more appropriate than "vendor supported", since we're only using the label with companies that have reached out to Chef's Partner Engineering team. The idea is that Partner Engineering will give vendors the guidance they need to write "better" cookbooks.

**Comment by martinb3** [comment]
:+1:

**Pull request closed without merging**